### PR TITLE
Add support for ~/!~ on single value queries

### PR DESCRIPTION
**Pull request by mwjames**
Allows to query things like `{{#ask: [[~Foo/*]] }}`

### CODE REVIEW

**Comment by mwjames** [comment]
http://sandbox.semantic-mediawiki.org/wiki/Issue/1246

File: src/SPARQLStore/QueryEngine/QueryEngine.php [hunk 1]
```diff
@@ -237,7 +238,7 @@ public function getDebugQueryResult( Query $query ) {
 
 		$entries = array();
 
-		if ( $sparqlCondition instanceof SingletonCondition ) {
+		if ( $sparqlCondition instanceof SingletonCondition && $sparqlCondition->matchElement instanceof Element ) {
```
Comment (JeroenDeDauw) [inline src/SPARQLStore/QueryEngine/QueryEngine.php]: You could extract this predicate

File: src/SQLStore/QueryEngine/Interpreter/ValueDescriptionInterpreter.php [hunk 2]
```diff
@@ -87,7 +87,11 @@ public function interpretDescription( Description $description ) {
 				$value
 			);
 
-			$query->where = "{$query->alias}.smw_sortkey$comparator" . $this->querySegmentListBuilder->getStore()->getConnection( 'mw.db' )->addQuotes( $value );
+			$db = $this->querySegmentListBuilder->getStore()->getConnection( 'mw.db' );
+
+			$query->where = "{$query->alias}.smw_sortkey$comparator" . $db->addQuotes( $value ).
+			" AND $query->alias.smw_iw!=" . $db->addQuotes( SMW_SQL3_SMWIW_OUTDATED ) .
```
Comment (JeroenDeDauw) [inline src/SQLStore/QueryEngine/Interpreter/ValueDescriptionInterpreter.php]: I'd put the concat symbol on the next line and use one more indent before it. This way its a lot easier to see at a glance what is happening

**Comment by Contrafibularity** [comment]
FWIW

A new tip has been added based on this splendid new feature, because it allows you to have queries for the previous and next results in a given category, concept, etc 

https://www.semantic-mediawiki.org/wiki/Help:Previous_and_next_pages_in_category
http://sandbox.semantic-mediawiki.org/wiki/Utilisateur:Cavila/previous_and_next

**Comment by kghbln** [comment]
Now [documended on wiki](https://www.semantic-mediawiki.org/w/index.php?title=Help%3ASelecting_pages&type=revision&diff=45377&oldid=45361).

**Pull request merged**